metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1814,7 +1814,14 @@ class MetalsLanguageServer(
       "started file watcher",
       clientConfig.initialConfig.statistics.isIndex
     ) {
-      fileWatcher.restart()
+      try {
+        fileWatcher.restart()
+      } catch {
+        // note(@tgodzik) This is needed in case of ammonite
+        // where it can rarely deletes directories while we are trying to watch them
+        case NonFatal(e) =>
+          scribe.warn("File watching failed, indexes will not be updated.", e)
```

Comment:
This might also be useful to log anyway just in case. There will be a fix in ammonite to not remove those directories, which will help in the long run.